hack/tools.sh

```diff
@@ -44,6 +45,60 @@ declare -r OC_URL="https://mirror.openshift.com/pub/openshift-v4/clients/ocp/$OC
 
 source "$PROJECT_ROOT/hack/utils.bash"
 
+go_install() {
+	local pkg="$1"
+	local version="$2"
+	shift 2
+
+	info "installing $pkg version: $version"
+
+	GOBIN=$LOCAL_BIN go install "$pkg@$version" || {
+		fail "failed to install $pkg - $version"
+		return 1
+	}
+	ok "$pkg - $version was installed successfully"
+	return 0
+}
+
+validate_version() {
+	local cmd="$1"
+	local version_arg="$2"
+	local version_regex="$3"
+	shift 3
+
+	command -v "$cmd" >/dev/null 2>&1 || return 1
+
+	[[ "$(eval "$cmd $version_arg" | grep -o "$version_regex")" =~ $version_regex ]] || {
+		return 1
+	}
+
+	ok "$cmd matching $version_regex already installed"
+}
+
+version_kubectl() {
+	kubectl version --client
+}
+
+install_kubectl() {
+	local version_regex="Client Version: $KUBECTL_VERSION"
+
+	validate_version kubectl "version --client" "$version_regex" && return 0
+
+	info "installing kubectl version: $KUBECTL_VERSION"
+	local install_url="https://dl.k8s.io/release/$KUBECTL_VERSION/bin/$GOOS/amd64/kubectl"
```

Comment:
Shouldn't we replace `amd64` with `$GOARCH`?

Reply:
Fixed